Automattic/sensei
# Add extensions installation feature

Based on https://github.com/Automattic/sensei/pull/4205

### Changes proposed in this Pull Request

* It introduces the installation logic to the extensions page.
* It doesn't handle edge cases, like WC not installed, not connected, and so on. (Next PRs)

In the future, we could think about making some of the queue items in parallel. For now, I think it's not worth the effort.

### Testing instructions

To simulate a pending update, you need to open the plugin main file and edit the header changing their version to an older one.


* Go to the extensions page.
* Install a WooCommerce extension.
* Install a free extension.
* Try installing and updating multiple extensions in parallel (clicking on multiple install/update buttons).
* Also, make a smoke test on Setup Wizard to make sure it's still installing the extensions properly.


### Screenshot / Video

https://user-images.githubusercontent.com/876340/117045896-6dd40880-ace6-11eb-8d1f-842a21f1fa48.mov
